hackathon_site/hardware/views.py

```diff
@@ -91,6 +86,22 @@ class OrderListView(generics.ListAPIView):
     ordering_fields = ("created_at",)
     search_fields = ("team__team_code", "id")
 
+    def get_queryset(self):
+        queryset = (
+            Order.objects.all().select_related("team")
+            # TODO: Causing problems with queryset aggregations, will figure out later:
+            # .prefetch_related("hardware", "hardware__categories")
+        )
+        if self.request.method == "GET":
+            if self.request.user.has_perm("hardware.view_order"):
+                return queryset
+            elif hasattr(self.request.user, "profile"):
+                return queryset.filter(team_id=self.request.user.profile.team_id)
+            else:
+                return queryset
```

Comment:
this is kind of messy but (I don't know why) I think get_queryset runs before get_permissions because for the test case where a user has no permissions or profile, django throws a 'User has no profile' exception. I added this check and the test now passes fine but I don't know how to fix this 🤔
@grahamhoyes @Leo6Leo

Reply:
Honestly I think this is fine.

I am starting to question our decision to have the APIs behave differently depending on your authentication level, though. Adding on to the discussion in #320, maybe it would have been cleaner to just leave this endpoint as admin-only and have something like `/api/event/teams/team/orders/` to return the current team's order.

The other problem that this brings up is that if we use this view on the participant side frontend to display their orders, it won't work properly if someone actually has order view permissions. Mostly that will happen when we are developing it locally, which could be really annoying.

What do you think @YuYing-Liang @Leo6Leo? We could keep this implicit filtering, or just start having dedicated endpoints for participants and admins.

Reply:
Hmmm you are right, I like the idea of separate endpoints and it prevents these weird permission issues

Reply:
Cool. Let's make this endpoint require just the Django model permissions then, and add a new one at the path above that filters by the current team. Can do that in this PR if you want, or make another one.